web/modules/custom/joinup_news/src/Controller/NewsController.php

```diff
@@ -64,14 +64,25 @@ public function add(RdfInterface $rdf_entity) {
   /**
    * Handles access to the news add form through rdf entity pages.
    *
+   * Access is granted to moderators and group members that have the permission
+   * to create news articles inside of their group, which in practice means this
+   * is granted to collection and solution facilitators.
+   *
    * @param \Drupal\rdf_entity\RdfInterface $rdf_entity
    *   The RDF entity for which the news entity is created.
    *
    * @return \Drupal\Core\Access\AccessResult
    *   The access result object.
    */
   public function createNewsAccess(RdfInterface $rdf_entity) {
-    return $this->ogAccess->userAccessGroupContentEntityOperations('create', $rdf_entity, $this->createNewsEntity($rdf_entity), $this->currentUser());
+    $user = $this->currentUser();
+    // Grant access if the user is a moderator.
+    if (in_array('moderator', $user->getRoles())) {
```

Comment:
In this case this is how it should be done in the current implementation of OG. `node_access_strict` doesn't work for creation of group content since `og_entity_access()` doesn't fire for it. Instead `og_entity_access_create()` fires but this doesn't have the correct group context and cannot decide whether a user has the rights to create group content because it doesn't know the groups.

I discussed this with @idimopoulos and we have created an issue in OG to deal with it: https://github.com/amitaibu/og/issues/272

I'll add links here and create a followup.